src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,212 @@
+    }
+
+    private static void specificTaskAdder(ArrayList<Task> recordedTask, String commandEntered) {
+        String[] taskCommandArr = commandEntered.split(" ", 2);
+        String exactDueDate = "";
+        try {
+            if (taskCommandArr.length <= 1) {
+                throw new DukeException();
+            } else if (taskCommandArr[1].contains("/by")) {
+                String taskDueDateString = taskCommandArr[1];
+                String[] taskDueDateArr = taskDueDateString.split("/by", 2);
+                exactDueDate = "(by:" + taskDueDateArr[1] + ")";
+                taskCommandArr[1] = taskDueDateArr[0];
+            } else if (taskCommandArr[1].contains("/at")) {
+                String taskDueDateString = taskCommandArr[1];
+                String[] taskDueDateArr = taskDueDateString.split("/at", 2);
+                exactDueDate = "(at:" + taskDueDateArr[1] + ")";
+                taskCommandArr[1] = taskDueDateArr[0];
+            }
+        } catch (DukeException de) {
+            printEmptyTaskErrorMessage(taskCommandArr[0]);
+            return;
+        }
+        switch (taskCommandArr[0]) {
+        case "todo":
+            recordedTask.add(new ToDos(Parser.getOrderAdded(), taskCommandArr[1], false, "T"));
+            break;
+        case "deadline":
+            recordedTask.add(new Deadlines(Parser.getOrderAdded(), taskCommandArr[1], false, "D"));
+            break;
+        case "event":
+            recordedTask.add(new Events(Parser.getOrderAdded(), taskCommandArr[1], false, "E"));
+            break;
+        default:
+            break;
+        }
+        recordedTask.get(Parser.getOrderAdded() - 1).setTaskName(taskCommandArr[1] + exactDueDate); //formats the output statement as desired
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(Messages.TASK_ADDER_AFFIRMATION);
+        recordedTask.get(Parser.getOrderAdded() - 1).printTaskListing();
+        if (Parser.getOrderAdded() == 1) {
+            System.out.println(Messages.TASK_ADDER_SINGULAR);
+        } else {
+            System.out.println(Messages.TASK_ADDER_PLURAL);
+        }
+        System.out.println(Messages.LINE_DIVIDER);
+        orderAdder();
+    }
+
+    private static void printEmptyTaskErrorMessage(String task) {
+        String statementPartOne = task.equals("event") ? Messages.EMPTY_TASK_ERROR_MESSAGE + "n ": Messages.EMPTY_TASK_ERROR_MESSAGE + " "; //an event vs a todo or deadline
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(statementPartOne + task + Messages.EMPTY_TASK_ERROR_MESSAGE_END);
+        System.out.println(Messages.LINE_DIVIDER);
+    }
+
+    private static void printUnrecognisedCommandErrorMessage() {
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(Messages.UNRECOGNISED_COMMAND);
+        System.out.println(Messages.LINE_DIVIDER);
+    }
+
+    private static void listOfTasksPrinter(ArrayList<Task> recordedTask) {
```

Comment:
Since recordedTask is an ArrayList of tasks, we can improve the naming convention used, so that it represents more than one task